core/kazoo_data/src/kzs_db.erl

```diff
@@ -222,9 +222,17 @@ add_update_remove_views(Server, Db, CurrentViews, NewViews, ShouldRemoveDangling
 add_views(Server, Db, Add, NewViews) ->
     Views = [props:get_value(Id, NewViews) || Id <- Add],
     {'ok', JObjs} = kzs_doc:save_docs(Server, Db, Views, []),
-    [kz_doc:id(JObj)
+    [Id
      || JObj <- JObjs,
-        <<"conflict">> =:= kz_json:get_value(<<"error">>, JObj)
+        Id <- [kz_doc:id(JObj)],
+        Error <- [kz_json:get_value(<<"error">>, JObj)],
+        'true' <- [(Error =/= 'undefined'
```

Comment:
Ugh, this is ugly. It is clever but hard to read, which I worry will make it hard to understand later.

Could you just make this a more explicit filter+log, something like:

```erlang
[Id || JObj <- JObjs, {Id, <<"conflict">>} <- [save_has_error(JObj)] ].

save_has_error(JObj) ->
    Error = kz_json:get_value(<<"error">>, JObj),
    Id = kz_doc:id(Id),
    save_has_error(Id, Error).

save_has_error(Id, <<"conflict">>=Error) -> {Id, Error};
save_has_error(Id, 'undefined'=Error) -> {Id, Error};
save_has_error(Id, Error) ->
    lager:warning("..."),
    {Id, Error}.
```

Something along these lines anyway. Make it more easy to read, less clever.